raiden/transfer/channel.py

```diff
@@ -443,52 +443,53 @@ def is_valid_lock_expired(
 ) -> MerkletreeOrError:
     secrethash = state_change.secrethash
     received_balance_proof = state_change.balance_proof
-    lock = channel_state.partner_state.secrethashes_to_lockedlocks.get(secrethash)
 
     # If the lock was not found in locked locks, this means that we've received
     # the secret for the locked transfer but we haven't unlocked it yet. Lock
     # expiry in this case could still happen which means that we have to make
     # sure that we check for "unclaimed" locks in our check.
+    lock = channel_state.partner_state.secrethashes_to_lockedlocks.get(secrethash)
     if not lock:
         lock = channel_state.partner_state.secrethashes_to_unlockedlocks.get(secrethash)
 
-    lock_registered_on_chain = (
-        secrethash in channel_state.our_state.secrethashes_to_onchain_unlockedlocks
+    secret_registered_on_chain = (
+        secrethash in channel_state.partner_state.secrethashes_to_onchain_unlockedlocks
     )
 
-    if not lock:
-        msg = (
-            f'Invalid LockExpired message. '
-            f'Lock with secrethash {pex(secrethash)} is not known.'
-        )
-        return (False, msg, None)
-
     current_balance_proof = get_current_balanceproof(sender_state)
-    merkletree = compute_merkletree_without(sender_state.merkletree, lock.lockhash)
-
     _, _, current_transferred_amount, current_locked_amount = current_balance_proof
-    expected_locked_amount = current_locked_amount - lock.amount
 
     is_balance_proof_usable, invalid_balance_proof_msg = is_balance_proof_usable_onchain(
         received_balance_proof=received_balance_proof,
         channel_state=channel_state,
         sender_state=sender_state,
     )
 
+    if lock:
+        merkletree = compute_merkletree_without(sender_state.merkletree, lock.lockhash)
+        expected_locked_amount = current_locked_amount - lock.amount
+
     result: MerkletreeOrError = (False, None, None)
 
-    if not is_balance_proof_usable:
+    if secret_registered_on_chain:
```

Comment:
lock will be `None` if the secret is registered on-chain, that is why I moved this check before the one just below